src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py

```diff
@@ -105,40 +106,73 @@ def prepare(self, corpus_dir, target_path, build_dir):
     # Directory corpus_dir saves the corpus files required by ClusterFuzz.
     arguments.append(f'--corpus_dir={corpus_dir}')
 
-    # The unsanitized binary, Centipede requires it to be the main fuzz target.
-    arguments.append(f'--binary={target_path}')
-
-    sanitized_target_path = self._get_sanitized_target_path(target_path)
-
-    if sanitized_target_path.exists():
-      arguments.append(f'--extra_binaries={sanitized_target_path}')
+    target_binaries = self._get_binary_paths(target_path)
+    if target_binaries.unsanitized is None:
+      # Assuming the only binary is always sanitized (e.g., from Chrome).
+      arguments.append(f'--binary={target_binaries.sanitized}')
+      logs.log_warn('Unable to find unsanitized target binary.')
     else:
-      logs.log_warn('Unable to find sanitized target binary.')
+      arguments.append(f'--binary={target_binaries.unsanitized}')
+      arguments.append(f'--extra_binaries={target_binaries.sanitized}')
 
     arguments.append(f'--timeout_per_input={_TIMEOUT_PER_INPUT_FUZZ}')
 
     arguments.extend(_DEFAULT_ARGUMENTS)
 
     return engine.FuzzOptions(corpus_dir, arguments, {})
 
-  def _get_sanitized_target_path(self, target_path):
-    """Get the path to the sanitized target based on the unsanitized target.
-    Sanitized targets are required by fuzzing (as an auxiliary) and crash
-    reproduction.
+  def _get_binary_paths(self, target_path):
+    """Gets the paths to the main and auxiliary binaries based on |target_path|
+    Args:
+      target_path: Path to the main target in a string.
+
+    Returns:
+      A named tuple containing paths to both target binaries as pathlib.Path.
+    """
+    # Centipede expects one or two target binaries:
+    # |-------------------------------------------------------|
+    # |            | main target path | auxiliary target path |
+    # |-------------------------------------------------------|
+    # | 1 binary   | sanitized        | -                     |
+    # |-------------------------------------------------------|
+    # | 2 binaries | unsanitized      | sanitized             |
+    # |-------------------------------------------------------|
+
+    main_target_path = pathlib.Path(target_path)
+    auxiliary_target_path = self._get_auxiliary_target_path(target_path)
```

Comment:
I feel like the term auxiliary_target doesn't really describe what's happening here if it's possible that there's only an auxiliary and not a main. Maybe just call it sanitized and unsanitized.

Reply:
I am a bit unsure if I understand you correctly: I don't think we will ever have an auxiliary without the main binary.
I intend to call the binary in the `BUILD_DIR` the main binary (which will always exist), and the one in `EXTRA_BUILD_DIR` the auxiliary (which may not exist).
They are not guaranteed to be sanitized or unsanitized:
1. If only the *main* binary exists, then we assume the main is sanitized. 
2. If both the *main* and the *auxiliary* exist, then we assume the main is unsanitized, and the auxiliary is sanitized.

I added this table to clarify this:
```
    3     # Centipede expects one or two target binaries:
    2     # |-------------------------------------------------------|
    1     # |            | main target path | auxiliary target path |
  136     # |-------------------------------------------------------|
    1     # | 1 binary   | sanitized        | -                     |
    2     # |-------------------------------------------------------|
    3     # | 2 binaries | unsanitized      | sanitized             |
    4     # |-------------------------------------------------------|

```

Reply:
Ah I misunderstood. This is fine!